bats/tests/k8s/traefik.bats

```diff
@@ -90,10 +95,44 @@ assert_traefik_pods_are_up() {
     wait_for_apiserver
     # Check if the traefik pods come up
     try --max 30 --delay 10 assert_traefik_pods_are_up
-    run try --max 30 --delay 10 curl --head "http://$(get_host):80"
+}
+
+@test 'curl traefik from distro' {
+    run try --max 30 --delay 10 curl --head "http://localhost:80"
     assert_success
     assert_output --regexp 'HTTP/[0-9.]* 404'
-    run try --max 30 --delay 10 curl --head --insecure "https://$(get_host):443"
+    run try --max 30 --delay 10 curl --head --insecure "https://localhost:443"
     assert_success
     assert_output --regexp 'HTTP/[0-9.]* 404'
 }
+
+@test 'curl traefik endpoints from Windows Host while kubernetes.ingress.localhost-only is false' {
+    if ! is_windows; then
+        skip "kubernetes.ingress.localhost-only is Windows only setting"
+    fi
+    curl_exe_http_traefik "http://$(wsl_ip):80"
+    curl_exe_https_traefik "https://$(wsl_ip):443"
+}
+
+@test 'set kubernetes.ingress.localhost-only to true' {
+    if ! is_windows; then
+        skip "kubernetes.ingress.localhost-only is Windows only setting"
+    fi
+    rdctl set --kubernetes.options.traefik=TRUE --kubernetes.ingress.localhost-only=TRUE
+    wait_for_apiserver
+    # Check if the traefik pods come up
+    try --max 30 --delay 10 assert_traefik_pods_are_up
+    assert_success
+}
+
+@test 'curl traefik endpoints from Windows Host while kubernetes.ingress.localhost-only is true' {
+    # traefik should only be accessible on localhost
+    curl_exe_http_traefik "http://localhost:80"
+    curl_exe_https_traefik "https://localhost:443"
+
+    # traefik should not be accessible on other interface
+    run curl.exe --head "http://$(wsl_ip):80"
+    assert_output --partial "Couldn't connect to server"
```

Comment:
I can't make up my mind if there should be an `assert_success` before testing `$output`, but I guess it doesn't matter, as we don't test how `curl` itself behaves.